datafusion/physical-plan/src/joins/utils.rs

```diff
@@ -955,7 +955,12 @@ fn max_distinct_count(
             let result = match num_rows {
                 Precision::Absent => Precision::Absent,
                 Precision::Inexact(count) => {
-                    Precision::Inexact(count - stats.null_count.get_value().unwrap_or(&0))
+                    // To safeguard against inexact number of rows (e.g. 0) being smaller than
+                    // an exact null count we need to do a checked subtraction.
+                    match count.checked_sub(*stats.null_count.get_value().unwrap_or(&0)) {
+                        None => Precision::Inexact(0),
```

Comment:
This made more sense than `Precision::Absent`.

Also, I'm not sure whether this can happen below as well, i.e. an inexact null count being larger than an exact row count.

Reply:
I agree -- the use of `Statistics::get_value()` I think may also have other bugs as `get_value()` may be exact or inexact but there are some places in the code that treat it as though it were always `exact` (like here)

I have hopes to improve statistics in general (see https://github.com/apache/arrow-datafusion/issues/8227) but other higher priority things have kept me busy. I think @berkaysynnada  was also working on this item for a while -- I am not sure if they have any short term plans

Reply:
This issue has been out of my focus for a while. I can help those who wish to take it on and make progress. Unfortunately, addressing this issue is not in my short-term plans.